evm/utils.js

```diff
@@ -642,57 +642,46 @@ const deepCopy = (obj) => JSON.parse(JSON.stringify(obj));
  * Retrieves and filters a list of EVM chains based on specified criteria.
  *
  * This function processes chain selection based on various input parameters and returns
- * a filtered list of chain objects that meet the specified criteria. It supports
- * case-insensitive chain name matching and validates that all chains are EVM-compatible.
+ * a filtered list of chain objects that meet the specified criteria.
  *
  */
 const getChains = (config, chainNames, skipChains, startFromChain) => {
-    const normalizeChainName = (name) => name.trim().toLowerCase();
-
     if (!chainNames) {
         throw new Error('Chain names were not provided');
     }
 
-    let chains = Object.entries(config.chains)
-        .filter(([_key, chain]) => !chain.chainType || chain.chainType === 'evm')
-        .map(([key, chain]) => [normalizeChainName(key), chain]);
-
-    let validChainNames = new Set(chains.map(([name, _chain]) => name));
+    const allChains = new Set(Object.keys(config.chains));
+    const parsedChainNames = chainNames === 'all' ? allChains : new Set(chainNames?.split(','));
+    const chainsToSkip = new Set(skipChains?.split(','));
 
-    if (startFromChain) {
-        const startIndex = chains.findIndex(([name, _chain]) => name === normalizeChainName(startFromChain));
-        if (startIndex === -1) {
-            throw new Error(`Chain ${startFromChain} is not in the selected chain list`);
+    parsedChainNames.forEach((name) => {
+        if (!allChains.has(name)) {
+            printError(`Chain to pick "${name}" is not defined in the config file`);
         }
-        chains = chains.slice(startIndex);
-    }
+    });
 
-    const normChainNames = new Set(chainNames?.split(',').map(normalizeChainName));
-    const chainsToSkip = new Set(skipChains?.split(',').filter(Boolean).map(normalizeChainName));
+    chainsToSkip.forEach((name) => {
+        if (!allChains.has(name)) {
+            printError(`Chain to skip "${name}" is not defined in the config file`);
```

Comment:
better to throw an exception. if there's a typo now we're not skipping a chain that the user intended to be skip which might be dangerous depending on the cmd being run